python/feathub/processors/tests/feathub_test_base.py

```diff
@@ -0,0 +1,157 @@
+#  Copyright 2022 The Feathub Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import shutil
+import tempfile
+import unittest
+import uuid
+from abc import abstractmethod
+from typing import Optional, List, Dict
+
+import pandas as pd
+
+from feathub.common import types
+from feathub.feathub_client import FeathubClient
+from feathub.feature_tables.sources.file_system_source import FileSystemSource
+from feathub.online_stores.memory_online_store import MemoryOnlineStore
+from feathub.table.schema import Schema
+
+
+class FeathubTestBase(unittest.TestCase):
+    """
+    Abstract base class for all processor integration tests. A child class of
+    this class must have its test cases use Feathub public APIs to get and write
+    features, and instantiate the corresponding FeathubClient instance.
+
+    This class also provides utility variables and methods to assist the construction
+    of test cases.
+    """
+
+    # By setting this attribute to false, it prevents pytest from discovering
+    # this class as a test when searching up from its child classes.
+    __test__ = False
+
+    def setUp(self) -> None:
+        self.temp_dir = tempfile.mkdtemp()
+        self.input_data, self.schema = self._create_input_data_and_schema()
+        self.client = self.get_client()
+
+    def tearDown(self) -> None:
+        MemoryOnlineStore.get_instance().reset()
+        shutil.rmtree(self.temp_dir, ignore_errors=True)
+
+    @abstractmethod
+    def get_client(self) -> FeathubClient:
+        """
+        Returns a FeathubClient instance for test cases.
+        """
+        pass
+
+    @staticmethod
+    def _get_local_client(processor_config: Dict) -> FeathubClient:
```

Comment:
Since this method can be called by subclasses, it seems better to rename this method as `get_local_client()`.

Same for other methods that can be called outside this class.